### PR TITLE
This is a simple PR to add a new TS type file for a koa library.  It follows the same format as other similar Koa middleware.

### PR DESCRIPTION
**Pull request by sjmcdowall**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

This WAS copied from a similar @types -- I tried to do it from scratch but couldn't get around the lint errors because I honestly have no idea of how to export default the function properly ..

### CODE REVIEW

File: types/koa-bunyan-logger/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Definitions by: Steven McDowall <https://github.com/sjmcdowall>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="node" />
+import { Middleware } from 'koa';
+
+import * as Logger from 'bunyan';
+
+export default function KoaBunyanLogger(logger: Logger | undefined): Middleware;
```
Comment (plantain-00) [inline types/koa-bunyan-logger/index.d.ts]: It's not `export default` https://github.com/koajs/bunyan-logger/blob/master/index.js#L261

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/koa-bunyan-logger/index.d.ts (hunk 1 above)
Comment (sjmcdowall) [reply types/koa-bunyan-logger/index.d.ts]: I'm sorry -- but I'm not a guru on converting from whatever form of module declaration is here to what TS should have.. 

What should : 
```
module.exports = function (loggerInstance) {
```
Get translated to in the index.d.ts file to be correct?  I'll happily make the change and re-submit. 

Thanks

**Comment by sjmcdowall** [comment]
OK -- I think I did what the reviewer wanted.  If not a bit more detail on what is desired...

File: types/koa-bunyan-logger/index.d.ts (hunk 1 above)
Comment (sjmcdowall) [reply types/koa-bunyan-logger/index.d.ts]: I think you are missing the idea of the module -- it's not the BUNYAN module we are writing the definition file for, it's the koa-bunyan-logger.  The key line is this:

https://github.com/koajs/bunyan-logger/blob/3ad9bef4faa1df8304ee6828989f2d22df465d27/index.js#L33

That is the core of the module .. The rest is "nice" but not the key bits .. 

As for a test file -- I honestly have no idea how to write a file for a piece of Koa Middleware and I don't see one Koa Middleware @types file with any tests to use as a sample..

Comment (plantain-00) [reply types/koa-bunyan-logger/index.d.ts]: Then
```js
function KoaBunyanLogger(...

export = KoaBunyanLogger;
```
For the test file, you can use the code from the document

```js
var koa = require('koa');
var koaBunyanLogger = require('koa-bunyan-logger');

var app = koa();
app.use(koaBunyanLogger());

app.use(function *(next) {
  this.log.info('Got a request from %s for %s', this.request.ip, this.path);
  yield next;
});

app.listen(8000);
```

**Comment by sjmcdowall** [comment]
I screwed up the git repo so started over and submitted a new PR ..

**Pull request closed without merging**